test/unit/test-http-handler.js

```diff
@@ -0,0 +1,209 @@
+    dispatch.resolves({ status: 200 });
+    getRawBodyStub.resolves(correctRawBody);
+    res.end.callsFake(function () {
+      assert(res.setHeader.calledWith('X-Slack-Powered-By'));
+      done();
+    });
+    this.middleware(req, res);
+  });
+
+  it('should handle application/json', function (done) {
+    var dispatch = this.dispatch;
+    var res = this.res;
+    var date = Math.floor(Date.now() / 1000);
+    var contentType = 'application/json';
+    var jsonBody = '{"type":"interactive_message","team":{"id":"T47563693"' +
+      ',"domain":"watermelonsugar"}}';
+    var req = createRequest(jsonBody, correctSigningSecret, date, contentType);
+    dispatch.resolves({ status: 200 });
+    getRawBodyStub.resolves(jsonBody);
+    res.end.callsFake(function () {
+      assert(dispatch.called);
+      assert.equal(res.statusCode, 200);
+      done();
+    });
+    this.middleware(req, res);
+  });
+
+  it('should respond to ssl check requests', function (done) {
+    var dispatch = this.dispatch;
+    var res = this.res;
+    var sslRawBody = 'payload=%7B%22ssl_check%22%3A%221%22%7D';
+    var date = Math.floor(Date.now() / 1000);
+    var contentType = 'application/x-www-form-urlencoded';
+    var req = createRequest(sslRawBody, correctSigningSecret, date, contentType);
+    getRawBodyStub.resolves(sslRawBody);
+    res.end.callsFake(function () {
+      assert(dispatch.notCalled);
+      assert.equal(res.statusCode, 200);
+      done();
+    });
+    this.middleware(req, res);
+  });
+
+  describe('handling dispatch results', function () {
+    it('should serialize objects in the content key as JSON', function (done) {
+      var dispatch = this.dispatch;
+      var res = this.res;
+      var date = Math.floor(Date.now() / 1000);
+      var contentType = 'application/x-www-form-urlencoded';
+      var req = createRequest(correctRawBody, correctSigningSecret, date, contentType);
+      var content = {
+        abc: 'def',
+        ghi: true,
+        jkl: ['m', 'n', 'o'],
+        p: 5
+      };
+      dispatch.resolves({ status: 200, content: content });
+      getRawBodyStub.resolves(correctRawBody);
+      res.end.callsFake(function (json) {
+        assert(dispatch.called);
+        assert.equal(res.statusCode, 200);
+        assert.deepEqual(json, JSON.stringify(content));
+        done();
+      });
+      this.middleware(req, res);
+    });
+
+    it('should handle an undefined content key as no body', function (done) {
+      var dispatch = this.dispatch;
+      var res = this.res;
+      var date = Math.floor(Date.now() / 1000);
+      var contentType = 'application/x-www-form-urlencoded';
+      var req = createRequest(correctRawBody, correctSigningSecret, date, contentType);
+      dispatch.resolves({ status: 500 });
+      getRawBodyStub.resolves(correctRawBody);
+      res.end.callsFake(function () {
+        assert(dispatch.called);
+        assert.equal(res.statusCode, 500);
```

Comment:
not that i care *too* much about this, but you should be asserting that `res.end()` is called with no argument (instead of a string).